cdm/core/src/main/java/ucar/nc2/internal/iosp/hdf4/H4header.java

```diff
@@ -104,7 +105,29 @@ public static void useHdfEos(boolean val) {
   private Map<Short, Vinfo> refnoMap = new HashMap<>();
 
   private MemTracker memTracker;
-  private PrintWriter debugOut = new PrintWriter(new OutputStreamWriter(System.out, StandardCharsets.UTF_8));
+  private PrintWriter debugOut;
+
+  private final Charset valueCharset;
+
+  public H4header() {
+    valueCharset = StandardCharsets.UTF_8;
+    debugOut = new PrintWriter(new OutputStreamWriter(System.out, StandardCharsets.UTF_8));
+  }
+
+  H4header(H4iosp h4iosp) {
+    valueCharset = h4iosp.getValueCharset().orElse(StandardCharsets.UTF_8);
+    debugOut = new PrintWriter(new OutputStreamWriter(System.out, valueCharset));
+  }
+
+  /**
+   * Return defined {@link Charset value charset} that
+   * will be used by reading netcdf file.
```

Comment:
The reference to netcdf file here isn't clear. In this case, it's the `Charset` used when reading string values out of the hdf4 header, right?